lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -65,16 +66,26 @@ def self.details
         "Release your beta builds with Firebase App Distribution"
       end
 
-      def self.platform
-        @platform ||= Actions.lane_context[Actions::SharedValues::PLATFORM_NAME]
+      def self.lane_platform
+        Actions.lane_context[Actions::SharedValues::PLATFORM_NAME]
+      end
+
+      def self.platform_from_path(binary_path)
+        return nil unless binary_path
+        case binary_path.split('.').last
+        when 'ipa'
+          :ios
+        when 'apk'
+          :android
+        end
       end
 
       def self.available_options
-        if platform == :ios || platform.nil?
+        if lane_platform == :ios || lane_platform.nil?
```

Comment:
Should we fall back `platform_from_path` here if `lane_platform` is nil as well?

Reply:
Unfortunately there's kind of a chicken-or-the-egg situation here, since `platform_from_path` depends on knowing the binary path, which is what we're trying to actually set here.

Luckily this block only comes into play if the user has not set an explicit path to the binary, and they also don't have a build block in the same lane, in which case we get the binary path from the lane context. So this should be an edge case.

If none of those are set, and neither is the platform, ideally we'd probably just look for both IPAs and APKs and use whichever one we find to determine the platform, but I think this is OK for now since again it should be kind of an edge case.

Reply:
Gotcha that makes sense. Thanks for the detailed explanation!